packages/delegation-process/src/services/delegationService.ts

```diff
@@ -60,23 +61,15 @@ import {
 } from "./validators.js";
 import { contractBuilder } from "./delegationContractBuilder.js";
 
-const retrieveDelegationById = async (
-  readModelService: ReadModelService,
-  delegationId: DelegationId
-): Promise<WithMetadata<Delegation>> => {
-  const delegation = await readModelService.getDelegation(delegationId);
-  if (!delegation?.data) {
-    throw delegationNotFound(delegationId);
-  }
-  return delegation;
-};
-
-export const retrieveDelegation = async (
+export const retrieveDelegationById = async (
   readModelService: ReadModelService,
   delegationId: DelegationId,
-  kind: DelegationKind
+  kind?: DelegationKind
```

Comment:
Could you keep this mandatory? It seems to be always specified...
Or, if it needs to be optional, use `DelegationKind | undefined` to force callers to explicitly set it.
I would prefer to have this mandatory to avoid the risk of retrieving a producer delegation in a consumer delegation call (e.g., consumer delegation approval) because we forget to specify the parameter
```suggestion
  kind: DelegationKind
```